app/response_processor.py

```diff
@@ -80,24 +78,12 @@ def process(self, msg, tx_id=None):
 
             decrypted_json['invalid'] = True
             self.logger.info("Invalid survey data, skipping receipting")
-        else:
-            if response_type.find("feedback") == -1:
-                self.logger.info("Receipting survey")
-                self.send_receipt(decrypted_json)
-            else:
-                self.logger.info("Feedback survey, skipping receipting")
 
         self.store_survey(decrypted_json)
 
-        if 'invalid' in decrypted_json:
-            self.logger.error(
-                "Invalid survey response received, skipping notification")
-
-        elif response_type.find("feedback") == -1:
-            self.send_notification(decrypted_json.get("survey_id"))
+        self.send_receipt(decrypted_json)
```

Comment:
Why do we call send_receipt every time, shouldn't we only call it when we're going to receipt instead of having some logic in send_receipt to avoid receipting?